common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationRequest.java

```diff
@@ -219,6 +227,15 @@ public B setWebViewZoomControlsEnabled(boolean webViewZoomControlsEnabled) {
         public abstract AuthorizationRequest build();
     }
 
+    /**
+     * Base64 encode the state parameter to make it URL safe.
+     *
+     * @return a string representing a Base64 encoded state parameter.
+     */
+    public String getEncodedState() {
```

Comment:
I'm wondering if rather than having this public accessor, if we should just set the state to encoded internally and keep using the same getState accessor

Reply:
Essentially we set `mState` to encodedUriSafeString version and then we don't need this method at all

Reply:
Yes, we can do that. 
I was essentially ensuring `setState` and `getState` receive and retrieve the same value respectively. But I agree it comes with some added complexity. 
I'll just set `mState` to the encodedUriSafeString and have javadoc in the methods.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1750/commits/49bf80b27b2e2c5a3e93047f563df5603050beb9